### PR TITLE
Converted to using omnisharp-node-client to control the server connectio...

### PR DESCRIPTION
**Pull request by david-driscoll**
...n.  Refactored to not wrap the client up too much and reduce events that are getting emitted.

Note: I fully expect this PR to around for a while :dash:

### CODE REVIEW

**Comment by david-driscoll** [comment]
Will require this to be completed at somepoint. (or move it to the npm binaries repo and complete it there).

 https://github.com/OmniSharp/omnisharp-roslyn/pull/186

**Comment by david-driscoll** [comment]
Alright, this should be good finally... rebased with the latest master.

**Comment by david-driscoll** [comment]
@nosami @ctolkien @willl @sgwill anyone have a chance to try this one out yet?

**Comment by sgwill** [comment]
Sorry, I'm at a wedding. I'll try tomorrow or Monday.

**Comment by david-driscoll** [comment]
Pfft Weddings :-1:  :smiling_imp:

**Comment by nosami** [comment]
OK.... Just tested this. Seems to be working well.

File: lib/omnisharp-atom/features/go-to-implementation.ts [hunk 1]
```diff
@@ -1,46 +1,42 @@
...
+
+    constructor(atomSharper: typeof OmniSharpAtom) {
+        this.atomSharper = atomSharper;
+    }
+
+    public goToImplementation() {
+        var editor = atom.workspace.getActiveTextEditor();
+        if (editor) {
+            var req: any = Omni.makeRequest();
+            req.word = <any>editor.getWordUnderCursor();
```
Comment (nosami) [inline lib/omnisharp-atom/features/go-to-implementation.ts]: what's this code for?

**Comment by david-driscoll** [comment]
cc @Mpdreamz

File: lib/omnisharp-atom/features/go-to-implementation.ts (hunk 1 above)
Comment (david-driscoll) [inline lib/omnisharp-atom/features/go-to-implementation.ts]: `findimplementationsPromise` ?

Comment (david-driscoll) [inline lib/omnisharp-atom/features/go-to-implementation.ts]: I see it now, yeah that's probably a bug, looking at it.  I'll fix it after dinner is done.  It's not attaching the word to the request if the symbol can't be found.

Comment (david-driscoll) [inline lib/omnisharp-atom/features/go-to-implementation.ts]: Updated.

**Comment by david-driscoll** [comment]
@ctolkien when you get back up tomorrow, try it again, I've made some changes that should move the server automatically inline with the client, and hopefully avoid that long path issue.

**Comment by ctolkien** [comment]
Progress!

**Pull request merged**